library/CloudBlobHelpers.cs

```diff
@@ -88,37 +86,31 @@ public static string GetContainerName(string blobPath)
             return containerName;
         }
 
-        public static ReplicatedTableReadBlobResult TryReadBlob<T>(CloudBlockBlob blob, out T configuration, out string eTag, Func<string, T> ParseBlobFunc)
+        public static ReplicatedTableReadBlobResult TryReadBlob<T>(BlobClient blob, out T configuration, out string eTag, Func<string, T> ParseBlobFunc)
             where T : class
         {
             configuration = default(T);
             eTag = null;
 
             try
             {
-                BlobRequestOptions options = new BlobRequestOptions()
-                {
-                    ServerTimeout = TimeSpan.FromSeconds(5),
```

Comment:
These timeouts were added in the past for a bug fix where the blob download was sometimes getting stuck

Reply:
Will make similar change to above for ServerTimeout. Will use cancellation token to have MaximumExecutionTimeout of 30 seconds.

```
string content;
using (StorageExtensions.CreateServiceTimeoutScope(TimeSpan.FromSeconds(5)))
{
    var cancellationTokenSource = new CancellationTokenSource();
    // Limiting Maximum execution time for all the requests to 30 seconds
    cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(30));
    content = blob.DownloadContent(cancellationTokenSource.Token).Value.Content.ToString();
}
```